pkg/awsutils/awsutils.go

```diff
@@ -701,27 +737,54 @@ func (cache *EC2InstanceMetadataCache) FreeENI(eniName string) error {
 	log.Debugf("Found eni %s attachment id: %s ", eniName, aws.StringValue(attachID))
 
 	// detach it first
-	v := true
 	detachInput := &ec2.DetachNetworkInterfaceInput{
 		AttachmentId: attachID,
-		Force:        aws.Bool(v),
+		Force:        aws.Bool(true),
 	}
 
 	start := time.Now()
-	_, err = cache.ec2SVC.DetachNetworkInterface(detachInput)
-	awsAPILatency.WithLabelValues("DetachNetworkInterface", fmt.Sprint(err != nil)).Observe(msSince(start))
-	if err != nil {
+	detachENIFunc := func() error {
+		if _, err := cache.ec2SVC.DetachNetworkInterface(detachInput); err != nil {
+			// If the eni is not found, then return no error, because we do not need to detach
+			if containsAttachmentIDNotFoundError(err) {
+				return nil
+			}
+			return err
+		}
+		return nil
+	}
+	if err := exponentialBackoff(detachENIFunc); err != nil {
 		awsAPIErrInc("DetachNetworkInterface", err)
 		log.Errorf("Failed to detach eni %s %v", eniName, err)
 		return errors.Wrap(err, "free eni: failed to detach eni from instance")
 	}
+	awsAPILatency.WithLabelValues("DetachNetworkInterface", fmt.Sprint(err != nil)).Observe(msSince(start))
```

Comment:
IDK how this is supposed to be used, but seems like we would want to record these within the backoff function?
I guess it really depends on what the purpose is, if its to know how long the actual api call took, then inside the block, if its how long the operation took to complete (with retries) then outside the block.

Reply:
You're totally right re: it's supposed to measure how long the API call took. For some reason, my brain saw it as how long the entire operation took to complete. Will update.